tools/agent_tool/src/cpp/main.cpp

```diff
@@ -0,0 +1,224 @@
+    if (parse.error())
+    {
+        option::printUsage(fwrite, stdout, usage, columns);
+        return 1;
+    }
+
+    if (options[optionIndex::HELP])
+    {
+        option::printUsage(fwrite, stdout, usage, columns);
+        return 0;
+    }
+
+    for (int i = 0; i < parse.optionsCount(); ++i)
+    {
+        option::Option& opt = buffer[i];
+        switch (opt.index())
+        {
+            case optionIndex::HELP:
+                // Not possible as this argument has been already handled and it exits the program
+                break;
+
+            case optionIndex::ENTITY_TYPE:
+                if (strcmp(opt.arg, CLIENT_ENTITY_KIND_ARG) == 0)
+                {
+                    entity_type = EntityType::CLIENT;
+                }
+                else if (strcmp(opt.arg, SERVER_ENTITY_KIND_ARG) == 0)
+                {
+                    entity_type = EntityType::SERVER;
+                }
+                else if (strcmp(opt.arg, REPEATER_ENTITY_KIND_ARG) == 0)
+                {
+                    entity_type = EntityType::REPEATER;
+                }
+                else
+                {
+                    std::cerr << "ERROR: incorrect entity type. Only <client|server|repeater> accepted." << std::endl;
+                    return 1;
+                }
+                break;
+
+            case optionIndex::CONNECTION_ADDRESS:
+                connection_address = std::string(opt.arg);
+                break;
+
+            case optionIndex::LISTENING_ADDRESS:
+                listening_address = std::string(opt.arg);
+                break;
+
+            case optionIndex::NAME:
+                strcpy(name, std::string(opt.arg).c_str());
+                break;
+
+            case optionIndex::TRANSPORT:
+                if (strcmp(opt.arg, "tcp") == 0)
+                {
+                    transport_protocol = eprosima::ddsrouter::core::types::TransportProtocol::tcp;
+                }
+                else if (strcmp(opt.arg, "udp") == 0)
+                {
+                    transport_protocol = eprosima::ddsrouter::core::types::TransportProtocol::udp;
+                }
+                break;
+
+            case optionIndex::UNKNOWN_OPT:
+                std::cerr << "ERROR: " << opt.name << " is not a valid argument." << std::endl;
+                option::printUsage(fwrite, stdout, usage, columns);
+                return 1;
+                break;
+        }
+    }
+
+    {
+        if (entity_type == CLIENT)
+        {
+            domain = connection_address;
+            logUser(AMLIPCPP_MANUAL_TEST, "Starting Manual Agent Node execution. Creating connection address...");
+        }
+        else
+        {
+            domain = listening_address;
+            logUser(AMLIPCPP_MANUAL_TEST, "Starting Manual Agent Node execution. Creating listening address...");
+        }
+
+        // Create connection address
+        auto address = eprosima::ddsrouter::core::types::Address(
+            12121,
```

Comment:
I'd change this to be able to choose the port. Otherwise NAT traversal is very restricted